src/libraries/System.Text.Json/tests/Serialization/ConstructorTests/ConstructorTests.Stream.cs

```diff
@@ -69,33 +74,38 @@ async Task RunTest<T>(string testData)
+            tasks[2] = Task.Run(async () => await RunTestAsync<Parameterized_Class_With_ComplexTuple>(Parameterized_Class_With_ComplexTuple.s_json));
             // JSON that doesn't bind to ctor args are matched with properties or ignored (as appropriate).
-            await RunTest<Person_Class>(Person_Class.s_json);
-            await RunTest<Person_Struct>(Person_Struct.s_json);
+            tasks[3] = Task.Run(async () => await RunTestAsync<Person_Class>(Person_Class.s_json));
+            tasks[4] = Task.Run(async () => await RunTestAsync<Person_Struct>(Person_Struct.s_json));
             // JSON that doesn't bind to ctor args or properties are sent to ext data if avaiable.
-            await RunTest<Parameterized_Person>(Parameterized_Person.s_json);
-            await RunTest<Parameterized_Person_ObjExtData>(Parameterized_Person_ObjExtData.s_json);
+            tasks[5] = Task.Run(async () => await RunTestAsync<Parameterized_Person>(Parameterized_Person.s_json));
+            tasks[6] = Task.Run(async () => await RunTestAsync<Parameterized_Person_ObjExtData>(Parameterized_Person_ObjExtData.s_json));
             // Up to 64 ctor args are supported.
-            await RunTest<Class_With_Ctor_With_64_Params>(Encoding.UTF8.GetString(Class_With_Ctor_With_64_Params.Data));
+            tasks[7] = Task.Run(async () => await RunTestAsync<Class_With_Ctor_With_64_Params>(Encoding.UTF8.GetString(Class_With_Ctor_With_64_Params.Data)));
             // Arg8deserialization honors attributes on matching property.
-            await RunTest<Point_MembersHave_JsonPropertyName>(Point_MembersHave_JsonPropertyName.s_json);
-            await RunTest<Point_MembersHave_JsonConverter>(Point_MembersHave_JsonConverter.s_json);
-            await RunTest<Point_MembersHave_JsonIgnore>(Point_MembersHave_JsonIgnore.s_json);
+            tasks[8] = Task.Run(async () => await RunTestAsync<Point_MembersHave_JsonPropertyName>(Point_MembersHave_JsonPropertyName.s_json));
+            tasks[9] = Task.Run(async () => await RunTestAsync<Point_MembersHave_JsonConverter>(Point_MembersHave_JsonConverter.s_json));
+            tasks[10] = Task.Run(async () => await RunTestAsync<Point_MembersHave_JsonIgnore>(Point_MembersHave_JsonIgnore.s_json));
             // Complex JSON as last argument works
-            await RunTest<Point_With_Array>(Point_With_Array.s_json);
-            await RunTest<Point_With_Dictionary>(Point_With_Dictionary.s_json);
-            await RunTest<Point_With_Object>(Point_With_Object.s_json);
+            tasks[11] = Task.Run(async () => await RunTestAsync<Point_With_Array>(Point_With_Array.s_json));
+            tasks[12] = Task.Run(async () => await RunTestAsync<Point_With_Dictionary>(Point_With_Dictionary.s_json));
+            tasks[13] = Task.Run(async () => await RunTestAsync<Point_With_Object>(Point_With_Object.s_json));
+
+            await Task.WhenAll(tasks);
```

Comment:
Same question; is there a benefit of introducing parallelism here?  Presumably it would make this test harder to debug. and it doesn't appear to be focused on concurrency issues.